client/my-sites/media-library/drop-zone.jsx

```diff
@@ -35,9 +39,13 @@ module.exports = React.createClass( {
 			return;
 		}
 
-		MediaActions.clearValidationErrors( this.props.site.ID );
-		MediaActions.add( this.props.site.ID, files );
-		this.props.onAddMedia();
+		this.props.onBeforeAddMedia( files )
+			.then(() => {
```

Comment:
>Will this error when `onBeforeAddMedia` is its default `noop` ?

Yes it does. The media modal has a separate drop zone implementation, and currently it errors when attempting to drop an image, thus redirecting the browser window to the image itself.